Tests/GRPCCoreTests/RPCErrorTests.swift

```diff
@@ -14,114 +14,151 @@
+      (Status.Code.unavailable, RPCError.Code.unavailable),
+      (Status.Code.dataLoss, RPCError.Code.dataLoss),
+      (Status.Code.unauthenticated, RPCError.Code.unauthenticated),
+    ]
+  )
+  func testErrorCodeFromStatusCode(statusCode: Status.Code, rpcErrorCode: RPCError.Code?) throws {
+    #expect(RPCError.Code(statusCode) == rpcErrorCode)
   }
 
+  @Test("Equatable Conformance")
   func testEquatableConformance() {
-    XCTAssertEqual(
-      RPCError(code: .cancelled, message: ""),
+    #expect(
       RPCError(code: .cancelled, message: "")
+        == RPCError(code: .cancelled, message: "")
     )
 
-    XCTAssertEqual(
-      RPCError(code: .cancelled, message: "message"),
+    #expect(
       RPCError(code: .cancelled, message: "message")
+        == RPCError(code: .cancelled, message: "message")
     )
 
-    XCTAssertEqual(
-      RPCError(code: .cancelled, message: "message", metadata: ["foo": "bar"]),
+    #expect(
       RPCError(code: .cancelled, message: "message", metadata: ["foo": "bar"])
+        == RPCError(code: .cancelled, message: "message", metadata: ["foo": "bar"])
     )
 
-    XCTAssertNotEqual(
-      RPCError(code: .cancelled, message: ""),
-      RPCError(code: .cancelled, message: "message")
+    #expect(
+      RPCError(code: .cancelled, message: "")
+        != RPCError(code: .cancelled, message: "message")
     )
 
-    XCTAssertNotEqual(
-      RPCError(code: .cancelled, message: "message"),
-      RPCError(code: .unknown, message: "message")
+    #expect(
+      RPCError(code: .cancelled, message: "message")
+        != RPCError(code: .unknown, message: "message")
     )
 
-    XCTAssertNotEqual(
-      RPCError(code: .cancelled, message: "message", metadata: ["foo": "bar"]),
-      RPCError(code: .cancelled, message: "message", metadata: ["foo": "baz"])
+    #expect(
+      RPCError(code: .cancelled, message: "message", metadata: ["foo": "bar"])
+        != RPCError(code: .cancelled, message: "message", metadata: ["foo": "baz"])
     )
   }
 
-  func testStatusCodeRawValues() {
-    for (code, expected) in Self.statusCodeRawValue {
-      XCTAssertEqual(code.rawValue, expected, "\(code) had unexpected raw value")
-    }
+  @Test(
+    "Status Code Raw Values",
+    arguments: [
+      (RPCError.Code.cancelled, 1),
+      (.unknown, 2),
+      (.invalidArgument, 3),
+      (.deadlineExceeded, 4),
+      (.notFound, 5),
+      (.alreadyExists, 6),
+      (.permissionDenied, 7),
+      (.resourceExhausted, 8),
+      (.failedPrecondition, 9),
+      (.aborted, 10),
+      (.outOfRange, 11),
+      (.unimplemented, 12),
+      (.internalError, 13),
+      (.unavailable, 14),
+      (.dataLoss, 15),
+      (.unauthenticated, 16),
+    ]
+  )
+  func testStatusCodeRawValues(statusCode: RPCError.Code, rawValue: Int) {
+    #expect(statusCode.rawValue == rawValue, "\(statusCode) had unexpected raw value")
+  }
+
+  @Test("Flatten causes with same status code")
+  func testFlattenCausesWithSameStatusCode() {
```

Comment:
This one needs updating now that `flatteningCauses` doesn't exist.

It'd also be good to have a test which verifies a cause with a different status code is given isn't flattened